app/src/main/java/com/hackadroid/pingu/com/hackadroid/pingu/activity/Intro3.java

```diff
@@ -0,0 +1,226 @@
+    private static final int MINIMUM_DISTANCE = 50; // 50m
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_intro3);
+
+        Button b = (Button) findViewById(R.id.b1);
+        b.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                log3.info("Inside Button click !");
+                Toast.makeText(getApplicationContext(), "button activity", Toast.LENGTH_LONG).show();
+                checkConnection();
+                findloc();
+                           }
+
+        });
+
+    }
+
+    private void checkConnection() {
+
+         /* GPS */
+        String bestprovider;
+        Criteria locationCriteria;
+        final LocationManager manager = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
+
+        Toast.makeText(getApplicationContext(), "Entering check", Toast.LENGTH_LONG).show();
+
+
+        locationCriteria = new Criteria();
+        locationCriteria.setAccuracy(Criteria.ACCURACY_COARSE);
+        locationCriteria.setAltitudeRequired(false);
+        locationCriteria.setBearingRequired(false);
+        locationCriteria.setCostAllowed(true);
+        locationCriteria.setPowerRequirement(Criteria.NO_REQUIREMENT);
+
+
+    /*    if (!manager.isProviderEnabled(LocationManager.GPS_PROVIDER)) {
+            Toast.makeText(getApplicationContext(), "GPS is disabled!Please turn on ", Toast.LENGTH_LONG).show();
+            Intent myIntent = new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS);
+            startActivity(myIntent);
+        } else {
+            Toast.makeText(getApplicationContext(), "GPS is enabled!", Toast.LENGTH_LONG).show();
+            if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+                // TODO: Consider calling
+                //    ActivityCompat#requestPermissions
+                // here to request the missing permissions, and then overriding
+                //   public void onRequestPermissionsResult(int requestCode, String[] permissions,
+                //                                          int[] grantResults)
+                // to handle the case where the user grants the permission. See the documentation
+                // for ActivityCompat#requestPermissions for more details.
+                return;
+            }
+            manager.requestLocationUpdates(providerName, 20000, 100, locationListener);
+        }*/
+
+        try {
+
+            GPSoff = Settings.Secure.getInt(getContentResolver(), Settings.Secure.LOCATION_MODE);
+
+        } catch (Settings.SettingNotFoundException e) {
+            e.printStackTrace();
```

Comment:
While catching an exception, pls add relevant log where it is failing along with printing the StackTrace so that it would be easy to Debug.